### ISSUE TITLE
CA1508 Reported when variables are swapped in a loop

### ISSUE THREAD

**Issue by yenneferofvengerberg** [dotnet/roslyn-analyzers#6249]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/nl-nl/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `'current.Count == 0' is always 'true'. Remove or refactor the condition(s) to avoid dead code.`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 6 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.402](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

The following code raises CA1508, when it should not. The following code is a naively stripped down version of a production code which exhibits the warning and is supposedly to loop forever (what it does). The analyzer suggests that `current.Count` is always zero, therefore the code should terminate immediately.

```cs
        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.ValueContentAnalysis)]
        [Trait(Traits.DataflowAnalysis, Traits.Dataflow.CopyAnalysis)]
        [Fact, WorkItem(6249, "https://github.com/dotnet/roslyn-analyzers/issues/6249")]
        public async Task SwappedTupleValueCompare_NoDiagnosticAsync()
        {
            await VerifyCSharpAnalyzerAsync(@"
using System.Collections.Generic;
using System.Diagnostics;

class Test
{
    public static List<T> Test1<T>(IEnumerable<T> source)
    {
        var current = new Queue<T>(source);
        var next = new Queue<T>();
        var result = new List<T>(current.Count);

        while (true)
        {
            while (current.TryDequeue(out var item))
            {
                next.Enqueue(item);
            }

            // swap with tuple
            (next, current) = (current, next);

            // swap with temp explictly
            //var t = current;
            //current = next;
            //next = t;

            Debug.Assert(next.Count == 0);

            // the following line is marked by CA1508
            if (current.Count == 0)
                return result;
        }
    }
}
");
        }
```

### Steps To Reproduce
1. Ensure this configuration is applied to the project
```xml
<Nullable>enable</Nullable>
<AnalysisLevel>latest-all</AnalysisLevel>
```
2. Invoke the code above `Test1(Enumerable.Range(0, 10));`

### Expected behavior
No CA1508 warning in the code

### Actual behavior
CA1508 is reported

## Additional context
This is a very difficult case, and I have no idea if a shortcut can be used to help the analyzer detect it, since this implies proving the termination of a program. Swapping manually does not resolve the problem anyway.
